nm: Fix profile name changing

cathay4t opened the pull request:
Previously, nmstate can not change profile name on existing connection.
This patch will update the profile name when `interface.profile-name`
been desired explicitly.

Integration test case included.

Resolves: https://issues.redhat.com/browse/RHEL-59239

liangwen12year commented on tests/integration/nm/profile_test.py:
```diff
@@ -654,3 +654,19 @@ def test_delete_mac_based_profile_using_profile_name(mac_based_profile_eth1):
+            }
+        ]
+    }
+    libnmstate.apply(desired_state)
```
Can we add the assertion to assert that the profile name is `eth1-new` ?

cathay4t replied:
Add assertion. Thanks.

liangwen12year commented:
Black command failed.

cathay4t committed:
nm: Fix profile name changing

Previously, nmstate can not change profile name on existing connection.
This patch will update the profile name when `interface.profile-name`
been desired explicitly.

Integration test case included.

Resolves: https://issues.redhat.com/browse/RHEL-59239

Signed-off-by: Gris Ge <fge@redhat.com>

cathay4t commented:
CI fixed. Please review the patch again. Thanks!

ffmancera reviewed (approved):
LGTM
